src/eos-autoupdater.c

```diff
@@ -249,8 +251,8 @@ update_stamp_file (guint64 last_successful_update_secs,
 
   if (randomized_delay_days > 0)
     {
-      gint32 actual_delay_days = g_random_int_range (0, randomized_delay_days + 1);
-      mtime.tv_sec += actual_delay_days * SEC_PER_DAY;
+      gint32 actual_delay_days = g_random_int_range (0, (gint32) randomized_delay_days + 1);
```

Comment:
I think this could be negative if for example the user sets the delay to `G_MAXUINT32 - 1`, although that shouldn't happen in practice. And I think it would only result in checking for updates more often than intended.

Reply:
`randomized_delay_days` is limited to [0, (`G_MAXINT32` / `SEC_PER_DAY`) - 1] on input, so should always be safe to convert to `gint32` and add 1 to.

Reply:
Oops, I missed that